wol/main.py

```diff
@@ -0,0 +1,184 @@
+
+Examples:
+  wol Ethernet10 00:11:22:33:44:55
+  wol Ethernet10 00:11:22:33:44:55 -b
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 00:22:44:66:88:aa
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 192.168.1.1 -c 3 -i 2000
+"""
+
+import binascii
+import click
+import copy
+import netifaces
+import os
+import socket
+import time
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+EPILOG = """\b
+Examples:
+  wol Ethernet10 00:11:22:33:44:55
+  wol Ethernet10 00:11:22:33:44:55 -b
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 00:22:44:66:88:aa
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 192.168.1.1 -c 3 -i 2000
+"""
+ORDINAL_NUMBER = ["0", "1st", "2nd", "3rd", "4th", "5th"]
+ETHER_TYPE_WOL = b'\x08\x42'
+
+
+class MacAddress(object):
+    def __init__(self, mac):
+        self.mac = mac
+
+    def __str__(self):
+        return "%x:%x:%x:%x:%x:%x" % (self.mac[0], self.mac[1], self.mac[2], self.mac[3], self.mac[4], self.mac[5])
+
+    def __eq__(self, other):
+        return self.mac == other.mac
+
+    def to_bytes(self):
+        return copy.copy(self.mac)
+
+
+BROADCAST_MAC = MacAddress(b'\xff\xff\xff\xff\xff\xff')
+
+
+def is_root():
+    return os.geteuid() == 0
+
+
+def get_interface_operstate(interface):
+    with open('/sys/class/net/{}/operstate'.format(interface), 'r') as f:
+        return f.read().strip().lower()
+
+
+def get_interface_mac(interface):
+    mac = netifaces.ifaddresses(interface)[netifaces.AF_LINK][0].get('addr')
+    return MacAddress(binascii.unhexlify(mac.replace(':', '').replace('-', '')))
+
+
+def build_magic_packet(interface, target_mac, broadcast, password):
+    dst_mac = BROADCAST_MAC if broadcast else target_mac
+    src_mac = get_interface_mac(interface)
+    return dst_mac.to_bytes() + src_mac.to_bytes() + ETHER_TYPE_WOL \
+        + b'\xff' * 6 + target_mac.to_bytes() * 16 + password
+
+
+def send_magic_packet(interface, pkt, count, interval, verbose):
+    target_mac = MacAddress(pkt[20:26])
+    if verbose:
+        print("Sending {} magic packet to {} via interface {}".format(count, target_mac, interface))
+    sock = socket.socket(socket.AF_PACKET, socket.SOCK_RAW)
+    sock.bind((interface, 0))
+    for i in range(count):
+        sock.send(pkt)
+        if verbose:
+            print("{} magic packet sent to {}".format(ORDINAL_NUMBER[i + 1], target_mac))
+        if i + 1 != count:
+            time.sleep(interval / 1000)
+    sock.close()
+
+
+def validate_interface(ctx, param, value):
+    if value not in netifaces.interfaces():
+        raise click.BadParameter("invalid SONiC interface name {}".format(value))
+    if get_interface_operstate(value) != 'up':
+        raise click.BadParameter("interface {} is not up".format(value))
+    return value
+
+
+def validate_target_mac(ctx, param, value):
+    mac_list = []
+    for mac_s in value.split(','):
+        try:
+            mac_b = binascii.unhexlify(mac_s.replace(':', '').replace('-', ''))
```

Comment:
what is the `_b` means?

Reply:
I used this suffix to describe the variable type is bytes. I've refactored the code of class `MacAddress`. No longer need this suffix now.